### ISSUE TITLE
Port boring-ssl tests

### ISSUE THREAD

**Issue by senier** [Componolit/componolit#116]

**Comment by senier** [comment]
The boringssl crytpo tests fail with a page fault for unknown reasons:

```
$ make run/test/boringssl
[...]
[init -> boringssl_crypto_test] Note: Google Test filter = -None
[init -> boringssl_crypto_test] [==========] Running 505 tests from 48 test cases.
[init -> boringssl_crypto_test] [----------] Global test environment set-up.
[init -> boringssl_crypto_test] [----------] 1 test from X509V3Test
[init -> boringssl_crypto_test] [ RUN      ] X509V3Test.NameTest
no RM attachment (READ pf_addr=0x28 pf_ip=0x80ef8 from pager_object: pd='init -> boringssl_crypto_test' thread='ep')
page fault, pd='init -> boringssl_crypto_test' thread='ep' cpu=0 ip=0x80ef8 address=0x28 stack pointer=0xa04fef18 qualifiers=0x4 irUwp reason=1
```

This happens to many other tests I've tried an may imply some fundamental issue in the setup. The fault address (`0x80ef8`) is within `ld.lib.so`, but the disassembly at that address makes no sense. To be investigated...

**Comment by senier** [comment]
This issue is a crash when invoking `pthread_rwlock_rdlock` in CRYPTO_STATIC_MUTEX_lock_read() (src/crypto/thread_pthread.c). We do not see a corresponding call to pthread_rwlock_init() - the address differs. The reason may be related to the strange definition of CRYPTO_MUTEX in ./src/include/openssl/thread.h. Only on MacOS they use a real pthread_rwlock_t. For now we disable threading by passing -DOPENSSL_NO_THREADS to the build.

**Comment by senier** [comment]
Some tests work now until the test suite fails with:

```
[init -> boringssl_crypto_test] [ RUN      ] X509Test.RSASign
[init -> boringssl_crypto_test] Warning: kill: kill not implemented
[init -> boringssl_crypto_test] Warning: kill: kill not implemented
[init] child "boringssl_crypto_test" exited with exit value 1

```

**Comment by senier** [comment]
The reason the test suite was failing was the missing random source. Configuring jitterentropy to provide `/dev/urandom` solved this issue. Almost all test succeed now. As we disabled threading, the threading tests need to be disabled, too. Some tests hang (or take a very long time):

* RSATest.GenerateFIP
* RSATest.RoundKeyLengths 
* BNTest.RandRange
* BNTest.PrimeChecking

`BIOTest.SocketConnect` is disabled as it tries to create an IPv6 socket which is unsupported.

Closing this boring issue.
